src/py/pyodide/http.py

```diff
@@ -433,10 +433,11 @@ async def pyfetch(url: str, **kwargs: Any) -> FetchResponse:
     else:
         signal = controller.signal
     kwargs["signal"] = signal
+    fetcher = kwargs["custom_fetcher"] if "custom_fetcher" in kwargs else _jsfetch
```

Comment:
Maybe let's just call the argument fetcher? And the `signal` handling could use a bit of tidying too. How about:
```py
async def pyfetch(url: str, /, *, signal: Any = None, fetcher = None, **kwargs: Any) -> FetchResponse:
    controller = AbortController.new()
    if signal:
        signal = abortSignalAny(Array.new(signal, controller.signal))
    else:
        signal = controller.signal
    kwargs["signal"] = signal
    fetcher = fetcher or _jsfetch
```
The docstring should also be updated to add this new parameter.